### PR TITLE
feat: Pre-releaser github workflow

### PR DESCRIPTION
**Pull request by alexlokshin-czi**

### CODE REVIEW

File: .github/workflows/prerelease.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+      - name: Prerelease
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --rm-dist -f .goreleaser.prerelease.yml
+        env:
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+          GORELEASER_CURRENT_TAG: ${{ steps.generate_tag.outputs.tag-name}}
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
```
Comment (edulop91) [inline .github/workflows/prerelease.yaml:62]: this block should live int he `release.yml` file and we can get rid of this file. 
then, for these steps, do them when we do not release (in that other action file)

Comment (alexlokshin-czi) [reply .github/workflows/prerelease.yaml:62]: On it

File: .goreleaser.prerelease.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  prerelease: true
+  github:
+    owner: chanzuckerberg
+    name: happy
+
+brews:
+  - description: "A command line tool for happy path."
+    tap:
+      owner: chanzuckerberg
+      name: homebrew-rc-tap
```
Comment (edulop91) [inline .goreleaser.prerelease.yml:26]:
```suggestion
      name: homebrew-tap
```

File: .goreleaser.prerelease.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - -trimpath
+
+release:
+  prerelease: true
+  github:
+    owner: chanzuckerberg
+    name: happy
+
+brews:
+  - description: "A command line tool for happy path."
```
Comment (edulop91) [inline .goreleaser.prerelease.yml:23]:
```suggestion
  - description: "A command line tool for happy path."
     name : "happy-rc"
```
https://goreleaser.com/customization/homebrew/

File: .github/workflows/prerelease.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
+      
+      - name: Generate PreRelease Version
+        id: generate_tag
+        run: |
+          VERSION=v$(cat version.txt)-$(git rev-parse --short HEAD)
```
Comment (edulop91) [inline .github/workflows/prerelease.yaml:50]: I think you also need to create/push the new tag

File: .goreleaser.prerelease.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    tap:
+      owner: chanzuckerberg
+      name: homebrew-rc-tap
+    homepage: "https://github.com/chanzuckerberg/happy"
+    test: system "#{bin}/happy"
```
Comment (edulop91) [inline .goreleaser.prerelease.yml:28]:
```suggestion
    test: system "#{bin}/happy-rc"
```

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -2,6 +2,14 @@ on:
...
     branches:
       - main
+  pull_request:
+    branches:    
+      - main
+    types:
+      - edited
+      - opened
+      - synchronize
+      - reopened
```
Comment (edulop91) [inline .github/workflows/release.yaml:12]:
```suggestion
```

File: .github/workflows/release.yaml [hunk 7]
```diff
@@ -43,12 +51,19 @@ jobs:
         # so we build the proper version
         with:
           fetch-depth: 0
-        if: ${{ steps.release.outputs.release_created }}
+        if: ${{ steps.release.outputs.release_created || github.event.release.prerelease }}
```
Comment (edulop91) [inline .github/workflows/release.yaml:54]:
```suggestion
```

File: .github/workflows/release.yaml [hunk 8]
```diff
@@ -43,12 +51,19 @@ jobs:
...
 
       - uses: actions/setup-go@v2
         with:
           go-version: 1.17
-        if: ${{ steps.release.outputs.release_created }}
+        if: ${{ steps.release.outputs.release_created || github.event.release.prerelease }}
```
Comment (edulop91) [inline .github/workflows/release.yaml:59]:
```suggestion
```

File: .github/workflows/release.yaml [hunk 9]
```diff
@@ -43,12 +51,19 @@ jobs:
...
+
+      - name: Generate Prerelease Version
+        id: generate_tag
+        run: |
+          VERSION=v$(cat version.txt)-$(git rev-parse --short HEAD)
+          echo "::set-output name=tag-name::$VERSION"
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
```
Comment (edulop91) [inline .github/workflows/release.yaml:66]:
```suggestion
        if: ${{ !steps.release.outputs.release_created }}
```

File: .github/workflows/release.yaml [hunk 10]
```diff
@@ -58,3 +73,20 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
         if: ${{ steps.release.outputs.release_created }}
+
+      - name: Tag snapshot
+        uses: tvdias/github-tagger@v0.0.1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          tag: ${{ steps.generate_tag.outputs.tag-name }}
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
```
Comment (edulop91) [inline .github/workflows/release.yaml:82]:
```suggestion
        if: ${{ !steps.release.outputs.release_created }}
```

File: .github/workflows/release.yaml [hunk 11]
```diff
@@ -58,3 +73,20 @@ jobs:
...
+
+      - name: Prerelease
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --rm-dist -f .goreleaser.prerelease.yml
+        env:
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+          GORELEASER_CURRENT_TAG: ${{ steps.generate_tag.outputs.tag-name }}
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
```
Comment (edulop91) [inline .github/workflows/release.yaml:92]:
```suggestion
        if: ${{ !steps.release.outputs.release_created }}
```

File: .goreleaser.prerelease.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+builds:
+  - binary: happy
```
Comment (edulop91) [inline .goreleaser.prerelease.yml:2]:
```suggestion
  - binary: happy-rc
```

**Review by edulop91 — APPROVED** [review]

**Pull request merged**